pkg/operation/botanist/migration.go

```diff
@@ -58,6 +60,50 @@ func (b *Botanist) runParallelTaskForEachComponent(ctx context.Context, componen
 	return flow.Parallel(fns...)(ctx)
 }
 
+// IsCopyOfBackupsRequired check if etcd backups need to be copied between seeds.
+func (b *Botanist) IsCopyOfBackupsRequired(ctx context.Context) (bool, error) {
+	// TODO: check if we can't trigger backup copy for shoots that were migrated before this was implemented and still have a backupentry
+	// in the backupbucket of a different seed. Could be done with a task annotation.
+	if b.Seed.GetInfo().Spec.Backup == nil || !b.isRestorePhase() {
+		return false, nil
+	}
+
+	// First we check if etcd-main etcd resource has been deployed. This is only true if backups have been copied.
+	if _, err := b.Shoot.Components.ControlPlane.EtcdMain.Get(ctx); client.IgnoreNotFound(err) != nil {
+		return false, err
+	} else if err == nil {
+		return false, nil
+	}
+
+	backupEntry, err := b.Shoot.Components.BackupEntry.Get(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// If the shoot's backupentry has not been switched to the new seed's backup bucket, then backup copying has not been started yet.
+	// And source backup entry has not been created.
+	if backupEntry.Spec.BucketName != string(b.Seed.GetInfo().UID) {
+		return true, nil
+	}
+
+	// TODO: Still have to figure out what to do if the shoot has already been migrated and we want to migrate back ot the original seed
+	// In that case the shoot's backupentry will already point to the proper bucket and no copying has to be done and no error should be
+	// returned in that case
+	sourceBackupEntry, err := b.Shoot.Components.SourceBackupEntry.Get(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// If the source backupentry exists, then the shoot's backupentry must have had its bucketName switched to the new seed and the
+	// source backupentry's bucketName must point to the old seed. Otherwise copy of backups is impossible and data loss will occur.
+	// TODO: Add a code to indicate that the error is non retriable and requires manual interaction.
```

Comment:
We agreed that even though it could not occur normally, but only due to manual intervention, this error is critical enough to warrant special handling.